lightning-invoice/src/payment.rs

```diff
@@ -173,8 +187,46 @@ where
 	logger: L,
 	event_handler: E,
 	/// Caches the overall attempts at making a payment, which is updated prior to retrying.
-	payment_cache: Mutex<HashMap<PaymentHash, usize>>,
-	retry_attempts: RetryAttempts,
+	payment_cache: Mutex<HashMap<PaymentHash, PaymentAttempts<T>>>,
+	retry: Retry,
+}
+
+/// Storing minimal payment attempts information required for determining if a outbound payment can
+/// be retried.
+#[derive(Debug,Clone,Copy)]
+struct PaymentAttempts<T: Time> {
+	/// This should be >= 1 as we only insert PaymentAttempts for a PaymentHash if there is at least
+	/// /// one attempt.
+	counts: usize,
+	#[cfg(feature = "std")]
+	/// This field is only used when retry is [`Retry::Timeout`] which is only build with feature std
+	first_attempted_at: T
```

Comment:
Hmm, I believe this leaves an unused parameter `T` in `no-std` builds. We either need to always include the field and let the `ConfiguredTime` "turn it off" in no-std or we need have two completely separate types for no-std vs std (which is gonna be a lot of code duplication).

Reply:
Then it's probably better to just always include this field(remove the configuration std condition check)?
Btw how can I replicate the error/warning of unused parameter `T` in `no-std` builds in local?(everything look fine when I do cargo build for no-std)